api/src/main/java/org/commonjava/indy/conf/DefaultIndyConfiguration.java

```diff
@@ -71,10 +75,49 @@
 
     private Boolean clusterEnabled;
 
+    private String nodeId;
+
     public DefaultIndyConfiguration()
     {
     }
 
+    @Override
+    public String getNodeId()
```

Comment:
Added nodeId configuration that we can use globally, which goes in main.conf under `indy.node.id` and can reference `${env.HOSTNAME}` or other expressions now. If unspecified, getDefaultNodeId() kicks in and tries to find a nodeId that's based on hostname (or, on windows, COMPUTERNAME).

Reply:
Do we expect to use the same main.conf for all nodes? So that we don't need to tweak it when starting a new node.

Reply:
I had expected to use an environment variable expression in here. Actually, I expected that most of the time we would leave it as some default value built into the system, but could use a different envar if we needed to by setting it here. The configuration changes in propulsor today enable use of envars like `${env.OPENSHIFT_POD}` if we need to use them.